QueryEngine/ConstantIR.cpp

```diff
@@ -96,38 +96,168 @@ llvm::ConstantInt* Executor::codegenIntConst(const Analyzer::Constant* constant)
   }
 }
 
+llvm::Value* find_arg_by_name(llvm::Function* func, const std::string& name) {
+  auto& arg_list = func->getArgumentList();
+  for (auto& arg : arg_list) {
+    if (arg.hasName() && arg.getName() == name) {
+      return &arg;
+    }
+  }
+  return nullptr;
+}
+
+
+llvm::Value* findHoistedValueInBlock(llvm::Function* function, std::string name) {
+  // WARNING: this does not work with the LICM optimizer... the generated IR seems correct, but it SIGSEGVs during peephole optimisation
+//  llvm::BasicBlock* hoistedVariablesBlock = find_BasicBlock_by_name(function, ".hoisted_variables");
+  llvm::BasicBlock* hoistedVariablesBlock = find_BasicBlock_by_name(function, ".entry");
+  CHECK(hoistedVariablesBlock);
+
+  llvm::BasicBlock& bb(*hoistedVariablesBlock);
+  // do a linear search for now
+  // should be changed to a binary search
+  for (llvm::Value& inst: bb) {
+    if (inst.hasName() && inst.getName() == name) {
+      return &inst;
+    }
+  }
+
+  return nullptr;
+}
+
 std::vector<llvm::Value*> Executor::codegenHoistedConstants(const std::vector<const Analyzer::Constant*>& constants,
                                                             const EncodingType enc_type,
                                                             const int dict_id) {
   CHECK(!constants.empty());
   const auto& type_info = constants.front()->get_type_info();
-  auto lit_buff_lv = get_arg_by_name(cgen_state_->row_func_, "literals");
-  int16_t lit_off{-1};
+  int64_t lit_off{-1};
```

Comment:
Could you describe why you decided to revamp this function? I see different versions.. What is the workflow?

Reply:
- codegenHoistedConstantsInPlace
  Creates the literal load from the literals buffer in place, effectively the original codegenHoistedConstants enhanced to also set the names for IR nodes accordingly
- codegenHoistedConstants 
  This is the original entry point, it has been changed to first check if the Constant was already hoisted (by checking presence in the function argument list). If not, it calls codegenHoistedConstantsInPlace to generate the load.
- codegenHoistedConstantsInEntryBlock (just renamed it)
  New entry point for the generation of the loads in the .entry block. It only creates the load, if the Constand had not been hoisted before.